libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/NSString+SFAdditions.m

```diff
@@ -213,10 +213,12 @@ - (nullable NSString*)entityId18 {
 }
 
 - (BOOL)isEqualToEntityId:(NSString*)entityId {
-    if ([self caseInsensitiveCompare:entityId] == NSOrderedSame) return YES; // for entityId like `me`
-    
-    if (![entityId length]) return NO;
-    if (!IsValidEntityId(self) || !IsValidEntityId(entityId)) return NO;
+    if (!IsValidEntityId(self) || !IsValidEntityId(entityId)) {
```

Comment:
If one or both are invalid ids, compare without looking at case (for the "me" - i guess we allow "me" and "Me" ??).